develop-docs/sdk/telemetry/logs.mdx

```diff
@@ -0,0 +1,381 @@
+    attributes: ["extra": @"123"]
+  )
+
+// Objective-C
+[SentrySDK.logger
+    emit :@"Adding item for user"
+    level: @"info"
+    attributes: @{ @"extra" : @"123", @"item_id" : item_id, @"user_id" : user_id }
+];
+```
+
+#### Threading and Concurrency Considerations
+
+Both the `Sentry.logger.X` and `Sentry.logger.emit` methods are fire-and-forget (have no return value). This means that the SDK can choose to run these methods on a separate thread or queue them up for processing later. This includes evaluating the string template, and running any internal hooks (like `beforeSendLog`). The SDK should ensure that the logs are sent in the order they are received.
+
+## SDK Behavior
+
+### Implementation
+
+If `debug` is set to `true` in SDK init, calls to the Sentry logger API should also print to the console with the appropriate log level. This will help debugging logging setups.
+
+Logs should be buffered before being sent. SDKs should keep a buffer of logs on the client (so you can have logs from multiple traces in the buffer) that flushes out based on some kind of condition. A recommended condition is if the buffer length exceeds 25 items, or if 5 seconds have passed. In the future we can use a [weight-tracking based approach](https://github.com/getsentry/rfcs/blob/main/text/0126-sdk-spans-aggregator.md#weight) to determine when to flush logs.
+
+The internal SDK implementation is kept purposefully broad as we are still in early stages. As we figure more out about the logs product and payload characteristics, we'll be able to define a more rigorous spec for logging internals in the SDK. SDKs should take care to not expose logging internals as much as possible to userland so that they can be refactored in the future.
+
+### Default Attributes
+
+By default the SDK should set the following attributes:
+
+1. `sentry.message.template`: The parameterized template string
+2. `sentry.message.parameters.X`: The parameters to the template string, where X is the number that represent the parameters position in the template string
+3. `sentry.environment`: The environment set in the SDK
+4. `sentry.release`: The release set in the SDK
+5. `sentry.trace.parent_span_id`: The span id of the span that was active when the log was collected. This should not be set if there was no active span.
+
+Example:
+
+```json
+{
+  "sentry.message.template": "Adding item %s for user %s",
+  "sentry.message.parameters.0": "item_id",
+  "sentry.message.parameters.1": "user_id",
+  "sentry.environment": "production",
+  "sentry.release": "1.0.0",
+  "sentry.trace.parent_span_id": "b0e6f15b45c36b12"
+}
+```
+
+Beyond these attributes, we are exploring if the SDK should also send OS, user, and device information automatically (via reading the appropriate contexts from the scope). Given this behaviour can easily be added as a new feature to the SDK, it does not have to be part of the initial SDK implementation until we make a finalized decision.
```

Comment:
Yes, I think we should. Especially if we want to replace breadcrumbs with logs.

Breadcrumbs are useful for understanding what happened before an error or a crash. Sending user and device info allows the product to query all logs happening before an error for a specific user or device. It would also allow you to view logs across app launches. Breadcrumbs are always only for one app launch only. For some problems, exploring what happened on a device on previous app launches could be useful.

Reply:
https://www.notion.so/sentry/DACI-Future-of-Breadcrumbs-once-we-have-Application-Logging-1618b10e4b5d8024bd01f6f2e232206d